main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1674,11 +1674,11 @@ object Typer {
           }
 
         for {
-          (ruleConstrs, ruleTypes, _, ruleEffs) <- traverseM(rules)(inferSelectRule).map(unzip4)
-          (defaultConstrs, defaultType, _, defaultEff) <- inferDefaultRule(default)
+          (ruleConstrs, ruleTypes, pur1, ruleEffs) <- traverseM(rules)(inferSelectRule).map(unzip4)
+          (defaultConstrs, defaultType, pur2, defaultEff) <- inferDefaultRule(default)
           resultCon = ruleConstrs.flatten ++ defaultConstrs
           resultTyp <- unifyTypeM(tvar :: defaultType :: ruleTypes, loc)
-          resultPur = Type.Impure
+          resultPur = Type.mkAnd(regionVar :: pur2 :: pur1, loc)
```

Comment:
-> `Type.mkAnd(pur1, pur2, regionVar, loc) `

Reply:
Just to check: this is a style thing, not a functionality thing (i.e., the meaning is the same as what I initially wrote, yes?

(just want to check I understand what's going on)

Reply:
Ah! I knew there was a reason why I wrote it the way I did: the form you suggest doesn't compile because `pur1` is of type `List[Type]` so I think it has to be the way I wrote it? Or is there another alternative?

Reply:
I've changed the variable name to hopefully make this clearer.